lib/v2/so/search.js

```diff
@@ -0,0 +1,48 @@
+const got = require('@/utils/got');
+const cheerio = require('cheerio');
+const { art } = require('@/utils/render');
+const path = require('path');
+const config = require('@/config').value;
+
+const renderDescription = (description, images) => art(path.join(__dirname, './templates/description.art'), { description, images });
+
+module.exports = async (ctx) => {
+    const { keyword } = ctx.params;
+    const url = `https://www.so.com/s?q=${encodeURIComponent(keyword)}`;
+    const key = `360-so-search:${url}`;
+    const items = await ctx.cache.tryGet(
+        key,
+        async () => {
+            const response = (await got(url)).data;
```

Comment:
This request requires cookie.
https://github.com/DIYgod/RSSHub/actions/runs/7309647602/job/19917459502#step:12:35

Reply:
> This request requires cookie. https://github.com/DIYgod/RSSHub/actions/runs/7309647602/job/19917459502#step:12:35

Thank you, I will fix this issue as soon as possible

Reply:
> cookie.

Hello, 
I have tried several ways to solve the issue, but I still cannot pass the GitHub Action request. 
I tested it locally and did not encounter the MaxRedirectsError or cookie problem. Even when accessing it through a proxy, I did not encounter this issue. 
Therefore, I suspect it is an IP issue with GitHub Action, and there may be some restrictions on so.com. 
If it doesn't work, I will temporarily remove the route to so.com to ensure that other changes can be merged in.